jpo-conflictmonitor/src/main/java/us/dot/its/jpo/conflictmonitor/monitor/topologies/MapSpatMessageAssessmentTopology.java

```diff
@@ -140,6 +139,12 @@ public Topology buildTopology() {
                                 event.setTimestamp(SpatTimestampExtractor.getSpatTimestamp(spat));
                                 event.setSpatRoadRegulatorIds(getAsSet(spat.getRegion()));
                                 event.setSpatIntersectionIds(getAsSet(spat.getIntersectionId()));
```

Comment:
Having independent sets on the IntersectionReferenceAlignmentEvent class to represent the MAP/SPAT road regulator ID and Intersection ID results in losing the association between the road regular ID and Intersection ID and also makes this code more complex than necessary.  Suggestion: refactor the `IntersectionReferenceAlignmentEvent` class.  Currently it looks like this:
```java
public class IntersectionReferenceAlignmentEvent extends Event{
    
    private String sourceID;
    private long timestamp;
    private Set<Integer> spatRoadRegulatorIds;
    private Set<Integer> mapRoadRegulatorIds;
    private Set<Integer> spatIntersectionIds;
    private Set<Integer> mapIntersectionIds;

...

}
```
Suggested change: Create a new class to encapsulate the regulator ID and intersection ID, something like:

```java
public class RegulatorIntersectionId {
    Integer roadRegulatorId;
    Integer intersectionId;
}
```
and refactor the event like:
```java
public class IntersectionReferenceAlignmentEvent extends Event{
    
    private String sourceID;
    private long timestamp;
    private Set<RegulatorIntersectionId> spatIds;
    private Set<RegulatorIntersectionId> mapIds;


}
```

Reply:
Good idea. Integrated!